modules/platforms/cpp/odbc/src/query/data_query.cpp

```diff
@@ -181,7 +181,11 @@ namespace ignite
             int64_t DataQuery::AffectedRows() const
             {
                 int64_t affected = rowsAffectedIdx < rowsAffected.size() ? rowsAffected[rowsAffectedIdx] : 0;
-                return affected < 0 ? 0 : affected;
+
+                if (affected >= 0)
+                    return affected;
+
+                return connection.GetConfiguration().GetPageSize();
```

Comment:
Just in case: different connections could yield different results for the same query.

Reply:
This is an expected behavior. Also, this is not a query, this is API call.